R/zchunk_L203.demand_input.R

```diff
@@ -88,288 +93,602 @@ module_aglu_L203.demand_input_DISABLED <- function(command, ...) {
+    # in all historical periods, regardless of which model base years are used. Processing data for all historical years
+    # that are also model years (i.e. not just the model base years)
+    aglu_demand_calyears <- HISTORICAL_YEARS[HISTORICAL_YEARS %in% MODEL_YEARS]
+    aglu_demand_futureyears <- c(max(aglu_demand_calyears), MODEL_YEARS[!MODEL_YEARS %in% aglu_demand_calyears])
+
+    # Define column names
+    # Supplysectors
+    names_Supplysector <- c( "region", "supplysector", "output.unit", "input.unit", "price.unit", "logit.year.fillout", "logit.exponent" )
+    names_SupplysectorLogitType <- c( "region", "supplysector", "logit.type" )
+    # Subsectors
+    names_Subsector <- c( "region", "supplysector", "subsector")
+    names_SubsectorAll <- c( names_Subsector, "logit.year.fillout", "logit.exponent", "year.fillout", "share.weight", "apply.to","from.year", "to.year", "interpolation.function" )
+    names_SubsectorLogitType <- c( names_Subsector, "logit.type" )
+    # Technologies
+    names_Tech <- c( "region", "supplysector", "subsector", "technology" )
+    names_StubTechProd <- c("region", "supplysector", "subsector", "stub.technology", "year", "calOutputValue", "share.weight.year", "subs.share.weight", "tech.share.weight")
+    names_StubTechFixOut <- c("region", "supplysector", "subsector", "stub.technology", "year", "fixedOutput", "share.weight.year", "subs.share.weight", "tech.share.weight")
+    names_StubTechCalorieContent <- c("region", "supplysector", "subsector", "stub.technology", "year", "minicam.energy.input", "efficiency", "market.name")
+    # Final demands
+    names_EnergyFinalDemand <- c("region", "energy.final.demand")
+    names_PerCapitaBased <- c(names_EnergyFinalDemand, "perCapitaBased")
+    Prod_colnames <- c("region", "supplysector", "year", "calOutputValue")
+    names_PriceElasticity <- c("region", "energy.final.demand", "year", "price.elasticity")
+
+    # L203.Supplysector_demand: generic info for demand sectors
+    A_demand_supplysector %>%
+      get_logit_fn_tables(names_SupplysectorLogitType, GCAM_region_names = GCAM_region_names,
+                          base_header = "Supplysector_", include_equiv_table = TRUE, write_all_regions = TRUE) ->
+      L203.SectorLogitTables
+    # Remove any regions for which agriculture and land use are not modeled
+    for(curr_table in names(L203.SectorLogitTables)) {
+      if(curr_table != "EQUIV_TABLE") {
+        L203.SectorLogitTables[[curr_table]]$data <- filter(L203.SectorLogitTables[[curr_table]]$data, !(region %in% aglu.NO_AGLU_REGIONS))
+      }
+    }
+
+    A_demand_supplysector %>%
+      write_to_all_regions(names_Supplysector, GCAM_region_names = GCAM_region_names) %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS) ->
+    L203.Supplysector_demand
+
+    # L203.SubsectorAll_demand: generic info for demand subsectors
+    A_demand_subsector %>%
+      get_logit_fn_tables(names_SubsectorLogitType, GCAM_region_names = GCAM_region_names,
```

Comment:
We need to figure out whether this should be here, or in the XML-generating file, because it returns a list of objects, not a tibble (which is what the driver is expecting). I _think_ we're going to want to move this logic elsewhere. Thoughts @pralitp ?